libraries/abstractions/ble_hal/test/src/iot_test_ble_hal_afqp.c

```diff
@@ -604,19 +613,42 @@ TEST( Full_BLE, BLE_Property_WriteCharacteristic )
  */
 TEST( Full_BLE, BLE_Property_WriteLongCharacteristic )
 {
-    prvWriteCheckAndResponse( bletestATTR_SRVCB_CHAR_A,
-                              true,
-                              true,
-                              0 );
-    prvWriteCheckAndResponse( bletestATTR_SRVCB_CHAR_A,
-                              true,
-                              true,
-                              bletestsMTU_SIZE1 - 5 );
-    /* /TODO: check what bNeedRsp should be (false here for CY) */
-    prvExecuteWriteCheckAndResponse( bletestATTR_SRVCB_CHAR_A,
-                                     false );
+    BLETESTwriteAttrCallback_t xWriteEvent;
+    BTStatus_t xStatus;
 
-    /* wait for execute write */
+    xStatus = IotTestBleHal_WaitEventFromQueue( eBLEHALEventWriteAttrCb, usHandlesBufferB[ bletestATTR_SRVCB_CHAR_A ], ( void * ) &xWriteEvent, sizeof( BLETESTwriteAttrCallback_t ), BLE_TESTS_WAIT );
+    TEST_ASSERT_EQUAL( eBTStatusSuccess, xStatus );
+    TEST_ASSERT_EQUAL( _usBLEConnId, xWriteEvent.usConnId );
+    TEST_ASSERT_EQUAL( 0, memcmp( &xWriteEvent.xBda, &_xAddressConnectedDevice, sizeof( BTBdaddr_t ) ) );
+    TEST_ASSERT_EQUAL( 0, xWriteEvent.usOffset );
+
+    if( xWriteEvent.bIsPrep == true )
+    {
+        TEST_ASSERT_EQUAL( usHandlesBufferB[ bletestATTR_SRVCB_CHAR_A ], xWriteEvent.usAttrHandle );
+        TEST_ASSERT_EACH_EQUAL_INT8( 49, xWriteEvent.ucValue, bletestsSTRINGYFIED_UUID_SIZE );
```

Comment:
Shouldn't we check for data fragment length instead of bletestsSTRINGYFIED_UUID_SIZE

Reply:
If you look into prvRequestWriteCb, you can see the callback funtion only keep the first bletestsSTRINGYFIED_UUID_SIZE bytes if data is longer than this length. So we can just check bletestsSTRINGYFIED_UUID_SIZE bytes rather than the actual data length.